trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/signing/SigningController.java

```diff
@@ -17,21 +17,19 @@
 import io.trino.aws.proxy.spi.credentials.Credential;
 import io.trino.aws.proxy.spi.credentials.Credentials;
 import io.trino.aws.proxy.spi.rest.Request;
+import io.trino.aws.proxy.spi.timestamps.AwsTimestamp;
 
 import java.net.URI;
-import java.time.Instant;
+import java.util.Optional;
 import java.util.function.Function;
 
 public interface SigningController
 {
-    String formatRequestInstant(Instant instant);
-
-    String formatResponseInstant(Instant instant);
-
-    String signRequest(
+    RequestAuthorization signRequest(
             SigningMetadata metadata,
             String region,
-            String requestDate,
+            AwsTimestamp requestDate,
```

Comment:
I'd prefer that we use JDK `Instant`s in any SPI APIs and convert internally to `AwsTimestamp`. `AwsTimestamp` is really an implementation detail and having it in SPI APIs feels like a leaky abstraction to me. wdyt?

Reply:
Agreed, happy to make the change